### PR TITLE
feat(frontend/backend): add possibility to delete own comments tm-489

### PR DESCRIPTION
**Pull request by tarmiel**

### CODE REVIEW

File: apps/backend/src/libs/hooks/check-is-user-creator/check-is-user-creator.hook.ts [hunk 1]
```diff
@@ -0,0 +1,36 @@
+import { type FastifyRequest } from "fastify";
+
+import { ExceptionMessage, HTTPCode } from "~/libs/enums/enums.js";
+import { type Service } from "~/libs/types/types.js";
+import { PermissionError } from "~/modules/permissions/permissions.js";
+
+type CheckIsUserCreatorParameters<T> = {
+	getDtoIdFromRequest: (request: FastifyRequest) => number;
+	getUserIdFromDto: (dto: T) => number;
```
Comment (what1s1ove) [inline apps/backend/src/libs/hooks/check-is-user-creator/check-is-user-creator.hook.ts:9]:
```suggestion
```
provide only paths. not cbs

**Review by what1s1ove — CHANGES_REQUESTED** [review]

File: apps/backend/src/modules/notifications/notification.service.ts [hunk 2]
```diff
@@ -87,10 +87,7 @@ class NotificationService implements Service {
 		);
 
 		if (!notification) {
-			throw new NotificationError({
-				message: ExceptionMessage.NOTIFICATION_NOT_FOUND,
-				status: HTTPCode.NOT_FOUND,
-			});
```
Comment (ShadowOfTheSpace) [inline apps/backend/src/modules/notifications/notification.service.ts:93]: Why did you remove exception here?

File: apps/backend/src/libs/hooks/check-is-user-creator/check-is-user-creator.hook.ts (hunk 1 above)
Comment (what1s1ove) [reply apps/backend/src/libs/hooks/check-is-user-creator/check-is-user-creator.hook.ts:9]: > What if we have nesting more than one level in dto? Should I handle parsing pathInDto: "author.id" into dto[author][id]

yes, you should handle it

> And what if we need to get the dto id not from params but from the query or body?

pass it as an argument of hook

File: apps/backend/src/libs/hooks/check-is-user-creator/check-is-user-creator.hook.ts [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+import { type Service } from "~/libs/types/types.js";
+import { PermissionError } from "~/modules/permissions/libs/exceptions/exceptions.js";
+
+type CheckIsUserCreatorParameters<T> = {
+	getDtoIdFromRequest: (request: FastifyRequest) => number;
+	getUserIdFromDto: (dto: T) => number;
+	service: Service<T>;
+};
+
+const checkIsUserCreator =
```
Comment (what1s1ove) [inline apps/backend/src/libs/hooks/check-is-user-creator/check-is-user-creator.hook.ts:13]: it is not only for user. make it more generic

File: apps/frontend/src/modules/activities/slices/activities.ts [hunk 4]
```diff
@@ -1,5 +1,6 @@
 import {
 	createComment,
+	deleteComment,
```
Comment (what1s1ove) [inline apps/frontend/src/modules/activities/slices/activities.ts:3]: it's strange that we have comments here. move all things that are related to comments to `comments` module

**Review by what1s1ove — CHANGES_REQUESTED** [review]

**Review by Sytnikovakatya — APPROVED** [review]

File: packages/shared/src/libs/helpers/get-value-by-path/get-value-by-path.helper.ts [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+	? Left extends keyof TData
+		?
+				| Extract<TData[Left], undefined>
+				| GetFieldType<Exclude<TData[Left], undefined>, Right>
+		: undefined
+	: TPath extends keyof TData
+		? TData[TPath]
+		: undefined;
+
+function getValueByPath<TData, TPath extends string>(
```
Comment (what1s1ove) [inline packages/shared/src/libs/helpers/get-value-by-path/get-value-by-path.helper.ts:11]: arrow function

File: apps/backend/src/libs/hooks/check-is-equal/check-is-equal.hook.ts [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+	}: CheckIsEqualParameters<DtoType, RequestType>) =>
+	async (request: FastifyRequest): Promise<void> => {
+		const toCompareId = getValueByPath(
+			request as RequestType,
+			pathInRequestToCompareId,
+		) as number;
+		const dtoId = getValueByPath(
+			request as RequestType,
+			pathInRequestToDtoId,
+		) as number;
```
Comment (what1s1ove) [inline apps/backend/src/libs/hooks/check-is-equal/check-is-equal.hook.ts:30]:
```suggestion
		const dtoId = getValueByPath<number>(
			request as RequestType,
			pathInRequestToDtoId,
		); // number or null
		
		if (!dtoId) {
		  // error
		}
```

File: apps/backend/src/libs/hooks/check-is-equal/check-is-equal.hook.ts [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+
+const checkIsEqual =
+	<DtoType, RequestType extends FastifyRequest>({
+		pathInDtoToCompareId,
+		pathInRequestToCompareId,
+		pathInRequestToDtoId,
+		service,
+	}: CheckIsEqualParameters<DtoType, RequestType>) =>
+	async (request: FastifyRequest): Promise<void> => {
+		const toCompareId = getValueByPath(
```
Comment (what1s1ove) [inline apps/backend/src/libs/hooks/check-is-equal/check-is-equal.hook.ts:23]:
```suggestion
		const compareId = getValueByPath(
```

File: apps/backend/src/libs/hooks/check-is-equal/check-is-equal.hook.ts [hunk 8]
```diff
@@ -0,0 +1,45 @@
...
+			request as RequestType,
+			pathInRequestToCompareId,
+		) as number;
+		const dtoId = getValueByPath(
+			request as RequestType,
+			pathInRequestToDtoId,
+		) as number;
+
+		const dto = await service.find(dtoId);
+		const dtoIdToCompare = getValueByPath(dto, pathInDtoToCompareId);
```
Comment (what1s1ove) [inline apps/backend/src/libs/hooks/check-is-equal/check-is-equal.hook.ts:33]:
```suggestion
		const compareDtoId = getValueByPath(dto, pathInDtoToCompareId);
```

File: apps/backend/src/libs/hooks/check-is-equal/check-is-equal.hook.ts [hunk 9]
```diff
@@ -0,0 +1,45 @@
...
+import { PermissionError } from "~/modules/permissions/libs/exceptions/exceptions.js";
+
+type CheckIsEqualParameters<DtoType, RequestType extends FastifyRequest> = {
+	pathInDtoToCompareId: ExtractPath<DtoType>;
+	pathInRequestToCompareId: ExtractPath<RequestType>;
+	pathInRequestToDtoId: ExtractPath<RequestType>;
+	service: Service<DtoType>;
+};
+
+const checkIsEqual =
```
Comment (what1s1ove) [inline apps/backend/src/libs/hooks/check-is-equal/check-is-equal.hook.ts:15]: do not like the name. should be something like this checkIsEqual<WHO> ?

File: apps/backend/src/libs/hooks/check-is-equal/check-is-equal.hook.ts (hunk 6 above)
Comment (what1s1ove) [reply apps/backend/src/libs/hooks/check-is-equal/check-is-equal.hook.ts:30]: same check for compareId

**Review by what1s1ove — CHANGES_REQUESTED** [review]

File: apps/backend/src/libs/hooks/check-is-equal/check-is-equal.hook.ts (hunk 9 above)
Comment (tarmiel) [reply apps/backend/src/libs/hooks/check-is-equal/check-is-equal.hook.ts]: I do not like it either, but I'm not completely sure what is the exact responsibility of this hook

**Review by what1s1ove — APPROVED** [review]

**Pull request merged**